File: examples/thrift-server/thrift-example-server/src/main/scala/com/twitter/calculator/CalculatorController.scala
package com.twitter.calculator

import com.twitter.calculator.thriftscala.Calculator
import com.twitter.calculator.thriftscala.Calculator._
import com.twitter.finatra.thrift.Controller
import com.twitter.util.Future
import javax.inject.Singleton

@Singleton
class CalculatorController
  extends Controller
  with Calculator.BaseServiceIface {

  override val addNumbers = handle(AddNumbers) { args: AddNumbers.Args =>
    info(s"Adding numbers $args.a + $args.b")
    Future.value(args.a + args.b)
  }

  override val addStrings = handle(AddStrings) { args: AddStrings.Args =>
    Future.value(
      (args.a.toInt + args.b.toInt).toString)
  }

  override val increment = handle(Increment) { args: Increment.Args =>
    Future.value(args.a + 1)
  }
}

File: examples/java-thrift-server/thrift-example-server/src/main/java/com/twitter/calculator/CalculatorController.java
package com.twitter.calculator;

import com.twitter.calculator.thriftjava.Calculator;
import com.twitter.util.Future;

public class CalculatorController implements Calculator.ServiceIface {

    @Override
    public Future<Integer> increment(int a) {
        return Future.value(a + 1);
    }

    @Override
    public Future<Integer> addNumbers(int a, int b) {
        return Future.value(a + b);
    }

    @Override
    public Future<String> addStrings(String a, String b) {
        Integer total = Integer.parseInt(a) + Integer.parseInt(b);
        return Future.value(total.toString());
    }
}
